Highway.cpp

```diff
@@ -13,22 +18,43 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if ( Car* c = dynamic_cast<Car*>(v) ) 
+    {
+        c->closeWindows();
+    } 
+    else if ( Motorcycle* mc = dynamic_cast<Motorcycle*>(v) )
+    {
+        mc->lanesplitAndRace(100);
+    }
+    else if ( SemiTruck* t = dynamic_cast<SemiTruck*>(v) )
+    {
+        t->honk();
+    }
 }
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that tries to evade the cops. 
 
     trucks pull over, but cars and bikes try to evade!!
     */
+    if ( Car* c = dynamic_cast<Car*>(v) ) 
+    {
+        c->tryToEvade();
+    } 
+    else if ( SemiTruck* t = dynamic_cast<SemiTruck*>(v) )
+    {
+        t->setSpeed(0);
```

Comment:
your `pullOver()` is already setting the speed.   
- [ ] remove this `setSpeed()` call